plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Debug/Debugger.hs

```diff
@@ -0,0 +1,267 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE ConstraintKinds #-}
+module UntypedPlutusCore.Evaluation.Machine.Cek.Debug.Debugger where
+
+import Annotation
+import UntypedPlutusCore
+import UntypedPlutusCore.Evaluation.Machine.Cek.Debug.Internal
+import Universe.Core
+
+import Data.RandomAccessList.Class qualified as Env
+import Data.Word64Array.Word8 hiding (Index, toList)
+import Control.Monad.State
+import Control.Monad.Trans.Free
+import Control.Lens hiding (Context)
+import Prettyprinter
+import Data.Sequence as Seq
+import Data.Coerce
+import GHC.Exts
+
+-- usually it is a line, but let's treat more generally as a srcspan
+newtype Breakpoint = Breakpoint { unBreakpoint :: SrcSpan }
+    deriving newtype (Show, Pretty)
+
+-- | The terms that we can debug
+type DTerm uni fun = Term NamedDeBruijn uni fun SrcSpans
+
+-- commands that the debugger can receive fromthe debug-client (tui,cli,test,etc)
+data Cmd
+  =
+  -- MAYBE: have a Load DTerm command; would require an initial invalid debugstate, i.e.  Maybe DebugState
+  Reload -- ^ reset back to start of initial term
+  -- small and big steps
+  | Step -- ^ step the cek once
```

Comment:
Given `StepF`, do you still need this `Step`?

Reply:
Yes, I think so.